src/app/components/plugins/Participation.tsx

```diff
@@ -0,0 +1,698 @@
+                "get",
+                undefined,
+                Participation.buildAuthHeaders());
+
+            if (response.data) {
+                this.setState({
+                    eventIds: response.data.eventIds
+                });
+            } else {
+                console.log(response.error);
+            }
+        } catch (err) {
+            console.log(err);
+        }
+    }
+
+    /**
+     *  Get the event information as a JSON payload.
+     * @param id Event id
+     */
+    private async fetchEventInfo(id: string): Promise<void> {
+        try {
+            const response = await FetchHelper.json<unknown, {
+                data?: IParticipationEventInfo;
+                error?: {
+                    message: string;
+                };
+            }>(
+                `${window.location.protocol}//${window.location.host}`,
+                `/api/plugins/participation/events/${id}`,
+                "get",
+                undefined,
+                Participation.buildAuthHeaders());
+
+            if (response.data) {
+                this.setState(prevState => ({
+                    events: {
+                        ...prevState.events,
+                        [id]: response.data as IParticipationEventInfo
+                    }
+                }));
+            } else {
+                console.log(response.error);
+            }
+        } catch (err) {
+            console.log(err);
+        }
+    }
+
+    /**
+     *  Get the event status info as a JSON payload.
+     * @param id Event id
+     */
+    private async fetchEventStatus(id: string): Promise<void> {
+        try {
+            const response = await FetchHelper.json<unknown, {
+                data?: IParticipationEventStatus;
+                error?: {
+                    message: string;
+                };
+            }>(
+                `${window.location.protocol}//${window.location.host}`,
+                `/api/plugins/participation/events/${id}/status`,
+                "get",
+                undefined,
+                Participation.buildAuthHeaders());
+
+            if (response.data) {
+                if (this.state.events[id]) {
+                    this.setState(prevState => ({
+                        ...prevState,
+                        events: {
+                            ...prevState.events,
+                            [id]: {
+                                ...prevState.events[id],
+                                status: response.data
+                            }
+                        }
+                    }));
+                }
+            } else {
+                console.log(response.error);
+            }
+        } catch (err) {
+            console.log(err);
+        }
+    }
+
+    /**
+     *  Submit event to be added.
+     */
+    private async submitEvent(): Promise<void> {
+        if (this.state.eventInfo) {
+            try {
+                const url = new URL(this.state.eventInfo);
```

Comment:
Should we maybe try to parse eventInfo as a valid JSON first, before doing requests?

Reply:
Changed to parse eventInfo as valid JSON first.